keystore/keystore.go

```diff
@@ -118,3 +142,56 @@ func AsListingKeystore(store Keystore) (ListingKeystore, error) {
 	}
 	return w, nil
 }
+
+// Factory Create the right keystore with the configured options.
+func Factory(c *config.C, defaultPath string, strictPerms bool) (Keystore, error) {
+	cfg := defaultConfig()
+
+	if c == nil {
+		c = config.NewConfig()
+	}
+	err := c.Unpack(&cfg)
+
+	if err != nil {
+		return nil, fmt.Errorf("could not read keystore configuration, err: %w", err)
+	}
+
+	if cfg.Path == "" {
+		cfg.Path = defaultPath
+	}
+
+	keystore, err := NewFileKeystoreWithStrictPerms(cfg.Path, strictPerms)
+	return keystore, err
+}
+
+// NewFileKeystore returns an new File based keystore or an error, currently users cannot set their
+// own password on the keystore, the default password will be an empty string. When the keystore
+// is initialized the secrets are automatically loaded into memory.
+func NewFileKeystore(keystoreFile string) (Keystore, error) {
+	return NewFileKeystoreWithStrictPerms(keystoreFile, false)
+}
+
+// NewFileKeystore returns an new File based keystore or an error, currently users cannot set their
+// own password on the keystore, the default password will be an empty string. When the keystore
+// is initialized the secrets are automatically loaded into memory.
+func NewFileKeystoreWithStrictPerms(keystoreFile string, strictPerms bool) (Keystore, error) {
+	return NewFileKeystoreWithPasswordAndStrictPerms(keystoreFile, NewSecureString([]byte("")), strictPerms)
+}
+
+// NewFileKeystoreWithPassword return a new File based keystore or an error, allow to define what
+// password to use to create the keystore.
+func NewFileKeystoreWithPassword(keystoreFile string, password *SecureString) (Keystore, error) {
+	return NewFileKeystoreWithPasswordAndStrictPerms(keystoreFile, password, false)
+}
+
+// randomBytes return a slice of random bytes of the defined length
+func randomBytes(length int) ([]byte, error) {
+	r := make([]byte, length)
+	_, err := rand.Read(r)
```

Comment:
can you do  `io.ReadFull(rand, r)` here? Read is not guaranteed to return exactly `length`.

Reply:
@mauri870 I do like the idea, do you meant to write `io.ReadFull(rand.Reader, randBytes)`? I actually like this proposal 🙂